src/hid-sensor.cpp

```diff
@@ -186,6 +186,7 @@ void hid_sensor::start( rs2_frame_callback_sptr callback )
     _hid_device->start_capture(
         [this, last_frame_number, last_timestamp]( const platform::sensor_data & sensor_data ) mutable
         {
+            std::lock_guard< std::mutex > lock( _configure_lock );
```

Comment:
This looks like a core function.
Do we really need to wrap all?
Can it impact the performance? or cause some deadlock?

Reply:
Since it only happens once every not so frequent call to start(), I believe there will be no performance impact.

We need to protect `_configured_profiles` from access by other threads during the call to 'start', which can happen if other threads calls `open` or 'close'. We can copy it to a local variable and have a smaller lock scope, but I think it might take more time and not really necessary, as the chances of simultaneous access are slim.